### DISCUSSION TITLE
Why Regolith Lockscreen (gnome-flashback?) is different than Gnome's?

### DISCUSSION THREAD

**Discussion by roxchgt** [regolith-linux/regolith-desktop#800] [Q&A] [answered]
Hi everyone,

I've been trying to figure out how I could use the Gnome's lock-screen also in Regolith? The lock screen is handled by gnome-flashback in Regolith which looks too simple (not important!) and doesn't support fingerprint authentication which is my main problem.

At the first login screen (pre-session) I can use my Thinkpad's fingerprint scanner to login instead of password, but in the session if the screen is locked, to get back in there is simple input field for password and no option for fingerprint, unless I choose the 'Switch User' button and get sent back to the initial login screen where I can choose other users as well as fingerprint authentication. However this doesn't happen when I use the Gnome desktop, the lock screen is same as the login (welcome) screen and everything works flawless!

I couldn't figure out how to handle the lock-screen in Regolith same as Gnome. I disabled the flashback as described here: https://regolith-desktop.com/docs/howtos/change-lockscreen/ but nothing replaces that, simply no lockscreen. could anyone help?

**Comment by kgilmer** [comment, marked as answer]
Hi @roxchgt , I don't have any specific ideas to try out.  But I think you might have some luck if you ask in the upstream `gnome-flashback` project.  I don't think there is anything in Regolith that changes the story...we just use what that project provides.

**Comment by dstathis** [comment]
@roxchgt Did you get an answer for this from upstream? Also, @kgilmer is there a way to use the builtin gnome lockscreen instead? It looks a bit nicer in addition to supporting fingerprint.

**Comment by dstathis** [comment]
Also @roxchgt I found a workaround [here](https://chedim.hashnode.dev/tip-unlocking-i3lock-with-fingerprint-scanner).

**Reply by roxchgt** [reply]
Thanks but I've gone without fingerprint, because there's still an old issue with fingerprint authentication and the Gnome keyring concept about which the community seems not to be able to agree upon! So it's kinda useless...

**Comment by kgilmer** [comment]
@dstathis , nice find on the i3lock fingerprint post.  Regarding your earlier question, you can override an Xresource key of `wm.program.lock` to specify any program you want to run in association w/ the lock keybinding.  More [details](https://github.com/regolith-linux/regolith-wm-config/blob/main/usr/share/regolith/i3/config.d/55_session_keybindings#L43)

**Reply by phillipprado** [reply]
Forgive my ignorance, but I still can't figure this out. I am also looking to replace the gnome-flashback lockscreen with the default gnome one, and I just can't seem to get it working. I've looked at your keybindings config file, I've tried everything I can think of in the Xresource file using wm.program.lock, and I just can't seem to get it working. What am I missing?

**Reply by roxchgt** [reply]
Up to this date, I coudn't find out `which` is the _default gnome login/lockscreen app_. I mean there must be a path to an executable and that is unknown, at least to me!

**Reply by kgilmer** [reply]
> default gnome one

This may be a feature of `gnome-shell`.  I'm not certain but if so, then you wouldn't be able use this as Regolith does not integrate with that.  If you can invoke the lockscreen of your choice from a terminal then specifying the path to it in `wm.program.lock` _should_ work.  If it doesn't have a look in `/var/log/messages` for recent related errors from `regolith-desktop`.

**Reply by edelmanjm** [reply]
If you look in `/usr/share/regolith/i3/config.d/55_session_keybindings`, youc an see it's sending a DBus call to Gnome:

```
dbus-send --type=method_call --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.Lock
```

That said, I have no idea how to get `gnome-flashback` to actually use fingerprints.
